demisto_sdk/commands/content_graph/interface/graph.py

```diff
@@ -15,48 +21,56 @@ def create_nodes(self, nodes: Dict[ContentType, List[Dict[str, Any]]]) -> None:
         pass
 
     @abstractmethod
-    def create_relationships(self, relationships: Dict[Relationship, List[Dict[str, Any]]]) -> None:
+    def create_relationships(self, relationships: Dict[RelationshipType, List[Dict[str, Any]]]) -> None:
         pass
 
     @abstractmethod
     def validate_graph(self) -> None:
         pass
 
-    @abstractmethod
-    def get_packs_content_items(self, marketplace: MarketplaceVersions):
-        pass
-
-    @abstractmethod
-    def get_all_integrations_with_commands(self):
-        pass
-
     @abstractmethod
     def clean_graph(self):
         pass
 
     @abstractmethod
-    def get_nodes_by_type(self, content_type: ContentType) -> Any:
-        pass
-
-    @abstractmethod
-    def search_nodes(
+    def search(
         self,
+        marketplace: MarketplaceVersions = None,
         content_type: Optional[ContentType] = None,
+        filter_list: Optional[Iterable[int]] = None,
+        all_level_dependencies: bool = False,
         **properties,
-    ) -> Any:
-        pass
+    ) -> List[Union[BaseContent, Command]]:
+        """
+        This searches the database for content items and returns a list of them, including their relationships
 
-    @abstractmethod
-    def get_single_node(
-        self,
-        content_type: Optional[ContentType] = None,
-        **properties,
-    ) -> Any:
-        pass
+        Args:
+            marketplace (MarketplaceVersions, optional): Marketplace to search by. Defaults to None.
+            content_type (Optional[ContentType], optional): The content_type to filter. Defaults to None.
+            filter_list (Optional[Iterable[int]], optional): A list of unique IDs to filter. Defaults to None.
+            all_level_dependencies (bool, optional): Whether to return all level dependencies. Defaults to False.
+            **properties: A key, value filter for the search. For example: `search(object_id="QRadar")`.
+
+        Returns:
+            List[Union[BaseContent, Command]]: The search results
+        """
+        if not marketplace and all_level_dependencies:
+            raise ValueError("Cannot search for all level dependencies without a marketplace")
+        # First we want to create pack dependencies if they are not availible
+        self.create_pack_dependencies()
+        return []
+
+    def marshal_graph(
+        self, marketplace: MarketplaceVersions
+    ) -> ContentDTO:
+        packs = self.search(
+            marketplace, content_type=ContentType.PACK
+        )
+        return ContentDTO(packs=packs)
 
     @abstractmethod
-    def get_relationships_by_type(self, relationship: Relationship) -> Any:
-        pass
+    def create_pack_dependencies(self):
+        ContentGraphInterface._with_dependencies = True
```

Comment:
why is this needed?

Reply:
If the graph is already has dependencies, we don't wan't to execute this query again

Reply:
let's talk about this too